### ISSUE TITLE
[import/no-unresolved] Not detecting certain Next.js components

### ISSUE THREAD

**Issue by shnparker** [benmosher/eslint-plugin-import#1856]
Hi, I'm trying to figure out why it's allowing certain jsx components to import fine (please let me know if more information is needed below). All mentioned components are .jsx

```jsx
// components/layout/PageLayout.jsx
import Head from "./Head";
import Nav from "./Nav";
import Footer from "./Footer";
```

but when I try and import next.js based pages, it cannot resolve

```jsx
//pages/index.jsx
import Head from "components/layout/Head"; (no-unresolved)
```
and another one

```jsx
// components/layout/PageLayout.jsx
import Head from "components/layout/Head"; (no-unresolved)
```

Here are my config fies
jsconfig.json
```json
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "pages/*": ["src/pages/*"],
      "components/*": ["src/components/*"],
      "hooks/*": ["src/hooks/*"],
      "styles/*": ["src/styles/*"],
      "utils/*": ["src/utils/*"]
    }
  }
}
```
.eslintrc.yml
```yaml
---
env:
  browser: true
  commonjs: true
  es6: true
  node: true
  jest: true
extends:
  - eslint:recommended
  - plugin:import/errors
  - plugin:import/warnings
  - plugin:react/recommended
  - plugin:jsx-a11y/recommended
  - prettier
  - prettier/react
globals:
  Atomics: readonly
  SharedArrayBuffer: readonly
parserOptions:
  sourceType: "module"
  ecmaVersion: 2020
  ecmaFeatures:
    jsx: true
plugins:
  - import
  - jest
  - react
  - react-hooks
settings:
  react:
    version: detect
  import/resolver:
    node:
      extensions:
        - .js
        - .jsx
    alias:
      - - "pages"
        - "./src/pages"
      - - "components"
        - "./src/components"
      - - "hooks"
        - "./src/hooks"
      - - "styles"
        - "./src/styles"
      - - "utils"
        - "./src/utils"
```

**Comment by shnparker** [comment]
After sleeping on it and playing with it a bit more, I realised that it was a problem with aliased paths that do not recognize .jsx. I have changed it to the following and that solved the problem:

```yaml
  import/resolver:
    alias:
      - - "pages"
        - "./src/pages"
      - - "components"
        - "./src/components"
      - - "hooks"
        - "./src/hooks"
      - - "styles"
        - "./src/styles"
      - - "utils"
        - "./src/utils"
      extensions:
        - .js
        - .jsx
```
